Server/JokesOnYou.Web.Api/Services/JokesService.cs

```diff
@@ -10,10 +14,68 @@ namespace JokesOnYou.Web.Api.Services
     public class JokesService : IJokesService
     {
         private readonly IJokesRepository _jokesRepo;
+        private readonly IUserRepository _userRepository;
+        private readonly IUnitOfWork _unitOfWork;
+        private readonly IMapper _mapper;
 
-        public JokesService(IJokesRepository jokesRepo)
+        public JokesService(IJokesRepository jokesRepo, IUserRepository userRepository, IUnitOfWork unitOfWork, IMapper mapper)
         {
             _jokesRepo = jokesRepo;
+            _mapper = mapper;
+            _userRepository = userRepository;
+            _unitOfWork = unitOfWork;
+        }
+        public async Task<JokeReplyDto> CreateJokeAsync(JokeCreateDto jokeCreateDto, string userId)
+        {
+            jokeCreateDto.Premise = jokeCreateDto.Premise.Trim();
```

Comment:
On the dto entity you can make use of the "set" and trim there. Im not sure if its a bad practice to trim tho considering this is the views job. Might wanna ask someone @PoisnFang

Reply:
This is just so all jokes that go into the database are trimmed of whitespace so we have as low of a duplicate as possible. (will not eliminate it of course )